### PR TITLE
batch synthesis samples

### PR DESCRIPTION
**Pull request by yulin-li**
## Purpose

Add samples for batch synthesis service

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Review by Ling-Cao — DISMISSED** [review]

File: samples/batch-synthesis/csharp/BatchSynthesisSample/Program.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    speechKey = keyValue;
+}
+else
+{
+    throw new ArgumentException($"Please set the SPEECH_KEY environment variable to set speech resource key.");
+}
+
+var host = $"https://{speechRegion}.customvoice.api.speech.microsoft.com";
+
+var sampleScript = "This is the waistline, and it's falling.\nAlexis, meet Bill and Hillary, and the rest of America.\nThis is Jordan, Scottie Pippen and the ring dynasty.\nThe more I looked, the gloomier I got.\nHe saw Macbeth, the three witches, and the boiling cauldron.\nBut it is good to get together, cook together, eat together and enjoy.\nThe priorities, they say, are sofa, blanket, and clothing.\nThe metallic walls curve, twist, and turn.\nThe seagrass fiber is tough, durable and smooth.\nThe dissenters were Stevens, Souter, Ginsburg and Breyer.";
```
Comment (eric-urban) [inline samples/batch-synthesis/csharp/BatchSynthesisSample/Program.cs:32]: Let's use Gatsby-chapter1.txt as the Python sample does.

File: samples/batch-synthesis/csharp/BatchSynthesisSample/BatchSynthesisClient.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        var batchSynthesis = new BatchSynthesis
+        {
+            DisplayName = displayName,
+            Description = description,
+            TextType = isSsml ? "Ssml" : "PlainText",
+            SynthesisConfig = new BatchSynthesisConfig
+            {
+                Voice = voiceName
+            },
+            Inputs = new List<BatchSynthesisInputDefinition> { new BatchSynthesisInputDefinition { Text = script } }
```
Comment (eric-urban) [inline samples/batch-synthesis/csharp/BatchSynthesisSample/BatchSynthesisClient.cs:93]: For documentation at learn.microsoft.com, does the max length of "Inputs" == 1,000? Each text input can have several paragraphs, but those paragraphs don't count against the 1,000 limit?

File: samples/batch-synthesis/csharp/BatchSynthesisSample/Program.cs (hunk 1 above)
Comment (yulin-li) [reply samples/batch-synthesis/csharp/BatchSynthesisSample/Program.cs:32]: good point. let me update

File: samples/batch-synthesis/csharp/BatchSynthesisSample/BatchSynthesisClient.cs (hunk 2 above)
Comment (yulin-li) [reply samples/batch-synthesis/csharp/BatchSynthesisSample/BatchSynthesisClient.cs:93]: Each job could has multi inputs at the maximum of 1000, i.e. the list length in this line <= 1000. We don't care the paragraph (i.e. line break) in each input script. And, the whole payload should < 500KB.

**Review by Ling-Cao — APPROVED** [review]

**Review by yinhew — APPROVED** [review]
sign-off

**Pull request merged**